packages/components/src/Combobox/Combobox.tsx

```diff
@@ -1,58 +1,91 @@
-import React from "react";
+import React, { useMemo } from "react";
 import { ComboboxProps } from "./Combobox.types";
 import { ComboboxContent } from "./components/ComboboxContent";
 import { ComboboxAction } from "./components/ComboboxAction";
 import { ComboboxContextProvider } from "./ComboboxProvider";
-import {
-  ComboboxTrigger,
-  ComboboxTriggerButton,
-  ComboboxTriggerChip,
-} from "./components/ComboboxTrigger";
-import { useComboboxValidation } from "./hooks/useComboboxValidation";
-import { ComboboxOption } from "./components/ComboboxOption";
+import { ComboboxTrigger } from "./components/ComboboxTrigger";
+import { ComboboxOption as ComboboxOptionComponent } from "./components/ComboboxOption/ComboboxOption";
+import styles from "./Combobox.css";
+import { useCombobox } from "./hooks/useCombobox";
 import { ComboboxActivator } from "./components/ComboboxActivator";
+import { useComboboxValidation } from "./hooks/useComboboxValidation";
 
-export function Combobox({
-  multiSelect = false,
-  label = "Select",
-  selected = [],
-  ...props
-}: ComboboxProps): JSX.Element {
-  const { contentElement, triggerElement, optionElements, actionElements } =
+export function Combobox(props: ComboboxProps): JSX.Element {
+  const { optionElements, triggerElement, actionElements } =
     useComboboxValidation(props.children);
 
-  const buildOptions = optionElements.map(option => ({
-    id: option.props.id,
-    label: option.props.label,
-  }));
+  const options = useMemo(
+    () =>
+      optionElements
+        ? optionElements.map(option => {
+            return {
+              id: option.props.id,
+              label: option.props.label,
+            };
+          })
+        : [],
+    [optionElements],
+  );
 
-  return (
-    <ComboboxContextProvider multiselect={multiSelect}>
-      {triggerElement || <ComboboxTrigger label={label} selected={selected} />}
+  const {
+    selectedOptions,
+    selectedStateSetter,
+    shouldScroll,
+    wrapperRef,
+    searchValue,
+    setSearchValue,
+    open,
+    setOpen,
+    handleClose,
+    handleSelection,
+  } = useCombobox(
+    props.selected,
+    props.onSelect,
+    props.onClose,
+    props.multiSelect,
+  );
 
-      {contentElement || (
-        // @ts-expect-error - Suppress the XOR error with onClose|onSelect until we finish the refactor on JOB-81416
-        <ComboboxContent options={buildOptions} selected={selected} {...props}>
-          {actionElements}
-        </ComboboxContent>
-      )}
+  return (
+    <ComboboxContextProvider
+      multiselect={props.multiSelect}
+      selected={selectedOptions}
+      selectionHandler={handleSelection}
+      open={open}
+      setOpen={setOpen}
+      handleClose={handleClose}
+      shouldScroll={shouldScroll}
+    >
+      <div ref={wrapperRef}>
```

Comment:
since buttons and chips are inline elements, gotta style this div to be inline-block instead of block, it'll always take up the whole row since it is wrapped in a block element